lalrpop/src/normalize/token_check/test.rs

```diff
@@ -163,6 +163,27 @@ fn match_catch_all() {
     assert!(validate_grammar(&grammar).is_ok())
 }
 
+/// Test that a `catch-all` can be use in the first `match` arm.
+/// Before the pull request to close [issue 325](https://github.com/lalrpop/lalrpop/issues/325),
+/// the usage of the `catch-all` symbol was not allowed in the first arm of a `match` block.
+fn match_catch_all_in_first_arm() {
+    let grammar = r#"
+        grammar;
+        match {
+            _
+        } else {
+            r"\w+"
+        }
+        pub Term = {
+            Num,
```

Comment:
let's add a case that overlaps with the `r"\w+"` that appears in the match. Maybe `r"[a-z]"`? This should not be an error, because it is given higher precedence.